installer/src/services/arduino-cli.js

```diff
@@ -48,7 +48,37 @@ const getBoards = ({
     });
 };
 
-const upload = ({ dstPath, fqbn, port, onData, onClose, onError }) => {
+const installRequiredCores = async ({ fqbnList, onData }) => {
+    const promises = [];
+    for (let i = 0; i < fqbnList.length; i += 1) {
+        const installRequiredCoresPromise = new Promise((resolve, reject) => {
```

Comment:
if I understand the code you are running multiple installations concurrently but you are returning the output with the same `onData` callback, so the output will be mixed. If that is the case I would prefer to have all the installation sequentially instead